homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,405 @@
+    traffic_mode = config[CONF_TRAFFIC_MODE]
+    route_mode = config[CONF_ROUTE_MODE]
+    name = config[CONF_NAME]
+    units = config.get(CONF_UNIT_SYSTEM, hass.config.units.name)
+
+    here_data = HERETravelTimeData(
+        here_client, travel_mode, traffic_mode, route_mode, units
+    )
+
+    sensor = HERETravelTimeSensor(name, origin, destination, here_data)
+
+    async_add_entities([sensor], True)
+
+
+def _are_valid_client_credentials(here_client: herepy.RoutingApi) -> bool:
+    """Check if the provided credentials are correct using defaults."""
+    test_origin = [38.9, -77.04833]
+    test_destination = [39.0, -77.1]
+    try:
+        here_client.car_route(
+            test_origin,
+            test_destination,
+            [
+                herepy.RouteMode[ROUTE_MODE_FASTEST],
+                herepy.RouteMode[TRAVEL_MODE_CAR],
+                herepy.RouteMode[TRAFFIC_MODE_DISABLED],
+            ],
+        )
+    except herepy.InvalidCredentialsError:
+        return False
+    return True
+
+
+class HERETravelTimeSensor(Entity):
+    """Representation of a HERE travel time sensor."""
+
+    def __init__(
+        self, name: str, origin: str, destination: str, here_data: "HERETravelTimeData"
+    ) -> None:
+        """Initialize the sensor."""
+        self._name = name
+        self._here_data = here_data
+        self._unit_of_measurement = UNIT_OF_MEASUREMENT
+        self._origin_entity_id = None
+        self._destination_entity_id = None
+
+        # Check if location is a trackable entity
+        if origin.split(".", 1)[0] in TRACKABLE_DOMAINS:
+            self._origin_entity_id = origin
+        else:
+            self._here_data.origin = origin
+
+        if destination.split(".", 1)[0] in TRACKABLE_DOMAINS:
+            self._destination_entity_id = destination
+        else:
+            self._here_data.destination = destination
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return the state of the sensor."""
+        if self._here_data.traffic_mode:
+            if self._here_data.traffic_time is not None:
+                return str(round(self._here_data.traffic_time / 60))
+        if self._here_data.base_time is not None:
+            return str(round(self._here_data.base_time / 60))
+
+        return None
+
+    @property
+    def name(self) -> str:
+        """Get the name of the sensor."""
+        return self._name
+
+    @property
+    def device_state_attributes(
+        self
+    ) -> Optional[Dict[str, Union[None, float, str, bool]]]:
+        """Return the state attributes."""
+        if self._here_data.base_time is None:
+            return None
+
+        res = {}
+        res[ATTR_ATTRIBUTION] = self._here_data.attribution
+        res[ATTR_DURATION] = self._here_data.base_time / 60
+        res[ATTR_DISTANCE] = self._here_data.distance
+        res[ATTR_ROUTE] = self._here_data.route
+        res[CONF_UNIT_SYSTEM] = self._here_data.units
```

Comment:
Units is static. Does it need to be part of the state attributes?

Reply:
It was requested by users to have it shown. I too find it nice to immediately see if the distance shown is yards, meter, kilometer...

I can add it as an parameter to have it as a static attribute of `HERETravelTimeSensor` itself. I need it in `HERETravelTimeData` to determine if I need to convert the units returned by the API.